aws/leftovers.go

```diff
@@ -62,7 +67,23 @@ func NewLeftovers(logger logger, accessKeyId, secretAccessKey, sessionToken, reg
 		Credentials: credentials.NewStaticCredentials(accessKeyId, secretAccessKey, sessionToken),
 		Region:      awslib.String(region),
 	}
-	sess := session.New(config)
+	sess, err := session.NewSession(config)
```

Comment:
This flow would have us recreate a session in the case that an assume role arn was passed so maybe we could switch to an if-else and only create the session once with whichever set of credentials are passed (i.e. accesskeyid+secretaccesskey+sessiontoken _or_ assumerolearn).

I can update the MR if you'd like with these changes.

Reply:
Oh apologies, I see that the initial session is used to create the credentials below after reading the docs for this: https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.44.260/aws/credentials/stscreds#hdr-Assume_Role

Reply:
Yes, you have to establish the initial session for the original account, then use that session to gather the credentials and establish a new session as the assumed role. The code could have been structured better to make that more obvious.

Thanks for the quick review and merge! 😄